contributor-agreement.md

```diff
@@ -0,0 +1,120 @@
+
+### 2\. License grant
+
+#### 2.1 Copyright license to Us
+
+Subject to the terms and conditions of this Agreement, You hereby grant to Us a worldwide, royalty-free, NON-exclusive, perpetual and irrevocable (except as stated in Section 8.2) license, with the right to transfer an unlimited number of non-exclusive licenses or to grant sublicenses to third parties, under the Copyright covering the Contribution to use the Contribution by all means, including, but not limited to:
+
+* publish the Contribution,
+* modify the Contribution,
+* prepare derivative works based upon or containing the Contribution and/or to combine the Contribution with other Materials,
+* reproduce the Contribution in original or modified form,
+* distribute, to make the Contribution available to the public, display and publicly perform the Contribution in original or modified form.
+
+#### 2.2 Moral rights
+
+Moral Rights remain unaffected to the extent they are recognized and not waivable by applicable law. Notwithstanding, You may add your name to the attribution mechanism customary used in the Materials you Contribute to, such as the header of the source code files of Your Contribution, and We will respect this attribution when using Your Contribution.
+
+### 3\. Patents
+
+3.1 Pledge of Identified Patents
+
+Whereas You are aware of certain of Your own patents, which You desire to identify and disclose as the following patents and patent applications ("Pledged Patents"):
+
+Number or Application Number  
+Country
+
+You hereby pledge to all developers, distributors and users of the Contribution or the Material to which this Agreement applies ("Pledge Recipient") that You and your Affiliates will not bring a lawsuit or other legal proceeding against a Pledge Recipient for patent infringement under any Pledged Patents based on the Pledge Recipient's making, having made, using, selling, offering for sale, importing or otherwise transferring the Contribution or the Material. You will require any person or entity to whom You sell or transfer any of the Pledged Patents to agree, in writing, to abide by the pledge and to place a similar requirement on any subsequent transferees to do the same.
+
+3.2 Revocation of Patent Pledge
+
+You reserve the right to revoke the patent pledge stated in section 3.1, as if it never existed with regard to any Pledge Recipient, who files a lawsuit or other legal proceedings for patent infringement or who has a direct financial interest in such a lawsuit or other legal proceedings against You or your Affiliates.
+
+### 4\. License obligations by Us
+
+We agree to license the Contribution only under the terms of the license or licenses that We are using on the Submission Date for the Material (including any rights to adopt any future version of a license).
+
+In addition, We may use the following licenses for Documentation in the Contribution: CC0-1.0 (including any right to adopt any future version of a license).
```

Comment:
Can we use the "No commitment to any specific outbound license" option?

Reply:
Hm, actually, I'm not 100% sure about this. Can leave as is.

Reply:
This is nice because it means that we could e.g. merge #5235 and all the similar ones without having to get permission.

Reply:
It does also mean we could pull a MongoDB and switch to a non-open source license.